src/test/java/uk/ac/ebi/ampt2d/accession/commons/generators/SingleAccessionGeneratorTest.java

```diff
@@ -0,0 +1,120 @@
+                                 Function<String, String> accessionHashingFunction) {
+        assertEquals("hash" + i, accessions.get(i).hash());
+        String accessionText = "test_name" + i + "_test_surname" + i;
+        if (accessionHashingFunction != null) {
+            accessionText = accessionHashingFunction.apply(accessionText);
+        }
+        assertEquals(accessionText, accessions.get(i).accession());
+        assertEquals("test_name" + i, accessions.get(i).model().name);
+        assertEquals("test_surname" + i, accessions.get(i).model().surname);
+    }
+
+    @Test
+    public void testOfHashAccession() {
+        SingleAccessionGenerator<TestUser, String> generator = SingleAccessionGenerator.ofHashAccessionGenerator(
+                testUser -> testUser.name + "_" + testUser.surname,
+                new SHA1HashingFunction()
+        );
+
+        HashMap<String, TestUser> hashToModel = new LinkedHashMap<>();
+        hashToModel.put("hash0", new TestUser("test_name0", "test_surname0"));
+        hashToModel.put("hash1", new TestUser("test_name1", "test_surname1"));
+        hashToModel.put("hash2", new TestUser("test_name2", "test_surname2"));
+
+        List<ModelHashAccession<TestUser, String, String>> accessions = generator.generateAccessions(hashToModel);
+        assertEquals(3, accessions.size());
+        assertAccession(0, accessions, new SHA1HashingFunction());
+    }
+
+    @Test
+    public void testOfSHA1HashAccession() {
+        SingleAccessionGenerator<TestUser, String> generator = SingleAccessionGenerator.ofSHA1AccessionGenerator(
+                testUser -> testUser.name + "_" + testUser.surname
+        );
+
+        HashMap<String, TestUser> hashToModel = new LinkedHashMap<>();
+        hashToModel.put("hash0", new TestUser("test_name0", "test_surname0"));
+        hashToModel.put("hash1", new TestUser("test_name1", "test_surname1"));
+        hashToModel.put("hash2", new TestUser("test_name2", "test_surname2"));
+
+        List<ModelHashAccession<TestUser, String, String>> accessions = generator.generateAccessions(hashToModel);
+        assertEquals(3, accessions.size());
+        assertAccession(0, accessions, new SHA1HashingFunction());
+    }
+
+    @Test
+    public void testPostSaveAction(){
+        SingleAccessionGenerator<TestUser, String> generator = SingleAccessionGenerator.ofSHA1AccessionGenerator(
+                testUser -> testUser.name + "_" + testUser.surname
+        );
+
+        HashMap<String, TestUser> hashToModel = new LinkedHashMap<>();
+        hashToModel.put("hash0", new TestUser("test_name0", "test_surname0"));
+        hashToModel.put("hash1", new TestUser("test_name1", "test_surname1"));
+        hashToModel.put("hash2", new TestUser("test_name2", "test_surname2"));
+
+        List<ModelHashAccession<TestUser, String, String>> accessions = generator.generateAccessions(hashToModel);
+        generator.postSave(new SaveResponse<>(hashToModel, new HashMap<>()));
```

Comment:
there is nothing that can be checked here? Just that the method is called and it does not produce exceptions?

Reply:
It is a do nothing function for this specific class.